### PR TITLE
Add support for suite and case IDs

### PR DESCRIPTION
**Pull request by emanchado**
We are going to add an id to test suites. For that to happen, however,
we first need to _allow_ it without requiring it, then change kpet-db
to add it, then change the property to be required.

This is the first step.

Ref. FASTMOVING-1278

### CODE REVIEW

**Review by spbnick — CHANGES_REQUESTED** [review]
This is certainly required, but not sufficient. To start modifying kpet-db you need this exposed to the templates, and that wouldn't be done unless you modify `kpet.run.Suite.__init__()`. Consider adding an integration test outputting a suite ID to Beaker XML, as well.

**Comment by spbnick** [comment]
Everything is good with this PR. However, we're getting ourselves in a pickle here. Cases already have "names", and those mean something different, actually they mean **two** different things. And I think it's high time we fixed that. Otherwise we're just adding more confusion. Let me think about it for a while.

**Comment by spbnick** [comment]
Added name defaulting to descriptions, and made descriptions optional, for smoother transition.
Added support for suite and case IDs as well, as that's what names where introduced for, and to avoid reviewing and testing a separate PR.

**Comment by spbnick** [comment]
@cki-bot, please test!

File: kpet/data.py [hunk 1]
```diff
@@ -314,17 +316,38 @@ def matches(self, target):
...
+
+    def validate_case_ids(self):
+        """
+        Check case IDs are unique.
+        """
+        id_set = set()
+        repeated_id_set = set()
+        for case in self.cases:
+            # pylint: disable=redefined-builtin, invalid-name
+            id = case.id
```
Comment (wizardofoz123) [inline kpet/data.py]: Shadows id()

File: kpet/data.py [hunk 2]
```diff
@@ -530,7 +560,23 @@ def get_sets_from_regex_list(regex_list):
                 if not case.sets <= suite.sets:
                     raise Invalid("Case sets are not a subset of suite sets "
                                   "in suite: {}\ncase: {}".
-                                  format(suite.description, case.name))
+                                  format(suite.name, case.name))
+
+    def validate_suite_ids(self):
```
Comment (wizardofoz123) [inline kpet/data.py:563]: This code is basically a duplicate of validate_case_ids(). I'd probably like to see these merged.

**Review by wizardofoz123 — CHANGES_REQUESTED** [review]
If possible, please see and fix the 2 comments I made fixed. Otherwise LGTM, test against kpet-db and merge.

File: kpet/data.py (hunk 2 above)
Comment (spbnick) [reply kpet/data.py:563]: Yeah, I thought about making a function returning a set of repeated members in a list, for example, but it wouldn't make this code much shorter. Do you still want me to get the common code out of these?

File: kpet/data.py (hunk 1 above)
Comment (spbnick) [reply kpet/data.py]: Yeah, I knooow :(. I think naming a function "id" was a bad move by Python developers. Grabbing such a common name. That's why I told pylint to bugger off. However, if you'd prefer I didn't do this, I can rename them to "case_id" and "suite_id".

Comment (wizardofoz123) [reply kpet/data.py]: Unintentionally shadowing anything is something I never do. Maybe that's a habit from C.
If you really want to, you can merge the code as-is. I'll leave that up to you.

File: kpet/data.py (hunk 2 above)
Comment (wizardofoz123) [reply kpet/data.py:563]: I'll leave that to you.

**Comment by spbnick** [comment]
Today I was less in a hurry and less lazy, so I went and extracted the common function, avoided clash with `id()` and added tests on top :D.

**Comment by spbnick** [comment]
@cki-bot, please test again!

**Comment by spbnick** [comment]
@cki-bot, please test, I promise I won't change it again :sob:!

**Review by wizardofoz123 — APPROVED** [review]

**Comment by spbnick** [comment]
All pipelines got to Beaker, merging!

**Pull request merged**